src/ports/postgres/modules/svm/kernel_approximation.py_in

```diff
@@ -0,0 +1,481 @@
+                   max_b=2*pi,
+                   row_dim=1,
+                   col_dim=self.n_components,
+                   rd_id=self.rd_id,
+                   rd_val=self.rd_val,
+                   random_state=self.random_state))
+
+        return self
+
+    def transform(self, source_table, independent_varname,
+                  dependent_varname=None, grouping_col=None, id_col=None,
+                  transformed_name='gaussian_transformed'):
+        if not self.rd_offset or not self.rd_weights:
+            return self
+
+        self.original_table = dict(source_table=source_table,
+                                   independent_varname=independent_varname,
+                                   dependent_varname=dependent_varname)
+
+        schema_madlib = self.schema_madlib
+
+        grouping_col = ("NULL::integer as {0}"
+                        .format(unique_string(desp='grouping_col'))
+                        if not grouping_col
+                        else "{0}".format(grouping_col))
+
+        dependent_varname = ("NULL::integer"
+                             if not dependent_varname
+                             else "{0}").format(dependent_varname)
+
+        id_col = ("NULL::integer as {0}"
+                  .format(unique_string(desp='id_col'))
+                  if not id_col
+                  else "{0}".format(id_col))
+
+        features_col = unique_string(desp='features_col')
+        target_col = unique_string(desp='target_col')
+        transformed = unique_string(desp=transformed_name)
+
+        # X = a * cos (X*C + b)
+        multiplier = sqrt(2. / self.n_components)
+        run_sql = """
+            drop table if exists {transformed};
+            create temp table {transformed} as
+                select
+                    {schema_madlib}.array_scalar_mult(
+                        {schema_madlib}.array_cos(
+                            {schema_madlib}.array_add(
+                                {schema_madlib}.__array_mdot(
+                                    q.{features_col}::float[],
+                                    rw.{val}::float[]
+                                )::float[],
+                                ro.{val}::float[]
+                            )::float[]
+                        )::float[],
+                        {multiplier}::float
+                    ) as {features_col},
+                    q.{target_col} as {target_col},
+                    {id_col},
+                    {grouping_col}
+                from (
+                    select
+                        {dependent_varname} as {target_col},
+                        {independent_varname} as {features_col},
+                        {id_col},
+                        {grouping_col}
+                    from {source_table}
+                ) q cross join (select {val} from {rd_weights}) as rw
+                    cross join (select {val} from {rd_offset}) as ro
+        """.format(id_col=id_col,
```

Comment:
To avoid the long format list (in multiple places) I would use the actual variable names in the format specifier and just use `.format(**locals())`